app/src/main/java/de/westnordost/streetcomplete/quests/amenity_indoor/AddIsAmenityIndoor.kt

```diff
@@ -0,0 +1,106 @@
+package de.westnordost.streetcomplete.quests.amenity_indoor
+
+import de.westnordost.osmfeatures.FeatureDictionary
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.geometry.ElementPolygonsGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.user.achievements.EditTypeAchievement.*
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.quests.YesNoQuestForm
+import de.westnordost.streetcomplete.util.ktx.toYesNo
+import de.westnordost.streetcomplete.util.math.isCompletelyInside
+import de.westnordost.streetcomplete.util.math.isInMultipolygon
+import java.util.concurrent.FutureTask
+
+class AddIsAmenityIndoor(private val featureDictionaryFuture: FutureTask<FeatureDictionary>) :
+    OsmElementQuestType<Boolean> {
+
+    private val nodesFilter by lazy {
+        """
+        nodes with
+         (
+         emergency ~ defibrillator|fire_extinguisher
+         or amenity ~ atm|telephone|parcel_locker|luggage_locker|locker|clock|post_box|public_bookcase|give_box|ticket_validator|vending_machine
+         )
+         and access !~ private|no
+         and !indoor and !location and !level and !level:ref
+    """.toElementFilterExpression()
+    }
+
+    /* We only want survey nodes within building outlines. */
+    private val BuildingFilter by lazy {
+        """
+    ways, relations with building
```

Comment:
I was not sure, if we should exclude private buildings like house,detached,etc. to make it less expensive to filter (less iterations of buildings.any { [...]isInMultipolygon }, but at the other hand, we could miss some amenitys at the outside wall  of these. I'm not sure, how often this would happen

Reply:
Any performance optimizations depend on whether the quest filter actually is slow. It is logged how long it took to process one quest.

Reply:
Ok, i tested this in two different areas. Is this performant enough?


**Suburban area with less of the selected amenitys**

AddIsAmenityIndoor: Found 4 quests in 1156ms
For comparance: Average for all quests was 240 ms, 8 quests took more time (AddHouseNumber,AddOpeningHours,CheckExistence,AddCycleway,AddTactilePavingKerb,
AddKerbHeight,AddMaxHeight,AddPathSurface)

<details>
<summary>Log for suburban area (23902 nodes, 4201 ways)</summary>
Downloaded 23902 nodes, 4201 ways and 38 relations in 12.6s
AddBusStopShelter: Found 0 quests in 331ms
AddBenchStatusOnBusStop: Found 0 quests in 404ms
AddSuspectedOneway: Found 0 quests in 490ms
AddBinStatusOnBusStop: Found 0 quests in 187ms
AddBusStopName: Found 0 quests in 114ms
AddPlaceName: Found 0 quests in 619ms
AddCarWashType: Found 0 quests in 44ms
AddTactilePavingBusStop: Found 0 quests in 253ms
AddAmenityCover: Found 0 quests in 24ms
AddBenchBackrest: Found 0 quests in 34ms
AddBusStopLit: Found 0 quests in 196ms
MarkCompletedBuildingConstruction: Found 2 quests in 56ms
AddBridgeStructure: Found 0 quests in 72ms
AddSport: Found 2 quests in 57ms
AddParkingType: Found 0 quests in 42ms
AddParkingAccess: Found 0 quests in 37ms
AddRailwayCrossingBarrier: Found 0 quests in 349ms
AddTrafficCalmingType: Found 0 quests in 9ms
AddPitchSurface: Found 10 quests in 241ms
AddPitchLit: Found 13 quests in 278ms
AddParkingFee: Found 0 quests in 210ms
AddStepsIncline: Found 0 quests in 22ms
AddTactilePavingSteps: Found 3 quests in 139ms
AddHandrail: Found 0 quests in 272ms
AddBicycleIncline: Found 0 quests in 166ms
AddHousenumber: Found 7 quests in 1249ms
AddMemorialType: Found 0 quests in 30ms
AddPowerPolesMaterial: Found 0 quests in 9ms
AddStepsRamp: Found 0 quests in 252ms
AddReligionToPlaceOfWorship: Found 0 quests in 30ms
AddReligionToWaysideShrine: Found 0 quests in 27ms
AddIsBuildingUnderground: Found 0 quests in 23ms
AddMotorcycleParkingCover: Found 0 quests in 28ms
AddMotorcycleParkingCapacity: Found 0 quests in 38ms
AddRecyclingType: Found 0 quests in 34ms
AddAirCompressor: Found 0 quests in 36ms
DetermineRecyclingGlass: Found 0 quests in 38ms
AddRecyclingContainerMaterials: Found 0 quests in 192ms
AddCrossingIsland: Found 2 quests in 277ms
AddOpeningHours: Found 21 quests in 1702ms
CheckExistence: Found 1 quests in 1827ms
AddCrossing: Found 0 quests in 553ms
AddTrafficSignalsButton: Found 0 quests in 9ms
AddCycleway: Found 0 quests in 1935ms
AddInformationToTourism: Found 0 quests in 26ms
AddPlaygroundAccess: Found 0 quests in 27ms
AddCrossingKerbHeight: Found 1 quests in 468ms
AddGritBinSeasonal: Found 0 quests in 12ms
AddBoardType: Found 0 quests in 6ms
AddBarrierType: Found 0 quests in 6ms
AddPostboxCollectionTimes: Found 0 quests in 233ms
AddCrossingType: Found 0 quests in 818ms
AddBicycleBarrierType: Found 0 quests in 6ms
AddBicycleBarrierInstallation: Found 0 quests in 19ms
AddTrafficSignalsVibration: Found 0 quests in 473ms
AddBarrierOnPath: Found 0 quests in 195ms
AddTrafficSignalsSound: Found 0 quests in 568ms
AddSeating: Found 8 quests in 106ms
AddBarrierOnRoad: Found 0 quests in 264ms
AddSelfServiceLaundry: Found 0 quests in 16ms
AddGeneralFee: Found 0 quests in 29ms
AddStileType: Found 0 quests in 196ms
AddCameraType: Found 0 quests in 5ms
AddDrinkingWaterType: Found 1 quests in 8ms
AddDrinkingWater: Found 0 quests in 22ms
AddTactilePavingCrosswalk: Found 6 quests in 961ms
AddFireHydrantType: Found 0 quests in 9ms
AddFireHydrantPosition: Found 0 quests in 22ms
AddFireHydrantDiameter: Found 0 quests in 23ms
AddBikeParkingCover: Found 2 quests in 16ms
AddBikeRentalType: Found 0 quests in 26ms
AddBikeParkingType: Found 2 quests in 20ms
AddBikeParkingAccess: Found 0 quests in 24ms
AddBikeParkingFee: Found 0 quests in 21ms
AddBollardType: Found 0 quests in 218ms
AddHairdresserCustomers: Found 4 quests in 13ms
SpecifyShopType: Found 0 quests in 14ms
AddBikeRentalCapacity: Found 0 quests in 30ms
AddTactilePavingKerb: Found 5 quests in 1329ms
AddAtmOperator: Found 0 quests in 4ms
AddBikeParkingCapacity: Found 2 quests in 50ms
AddAtmCashIn: Found 0 quests in 10ms
AddBicyclePump: Found 0 quests in 31ms
AddKerbHeight: Found 5 quests in 1367ms
AddAccessPointRef: Found 0 quests in 7ms
AddChargingStationOperator: Found 0 quests in 19ms
AddChargingStationCapacity: Found 0 quests in 36ms
AddWheelchairAccessOutside: Found 0 quests in 17ms
AddMaxWeight: Found 0 quests in 15ms
AddRoadName: Found 0 quests in 18ms
AddStepCountStile: Found 0 quests in 377ms
AddClothingBinOperator: Found 0 quests in 112ms
AddFerryAccessPedestrian: Found 0 quests in 9ms
AddFerryAccessMotorVehicle: Found 0 quests in 12ms
AddProhibitedForPedestrians: Found 0 quests in 17ms
MarkCompletedHighwayConstruction: Found 0 quests in 11ms
AddOneway: Found 0 quests in 200ms
AddSummitCross: Found 0 quests in 117ms
CheckOpeningHoursSigned: Found 0 quests in 321ms
AddOrchardProduce: Found 0 quests in 19ms
AddMaxPhysicalHeight: Found 0 quests in 280ms
AddAddressStreet: Found 0 quests in 418ms
AddSummitRegister: Found 0 quests in 168ms
CheckShopType: Found 0 quests in 506ms
AddAirConditioning: Found 9 quests in 177ms
AddLevel: Found 0 quests in 214ms
AddWheelchairAccessPublicTransport: Found 0 quests in 99ms
AddEntrance: Found 1 quests in 454ms
AddCampDrinkingWater: Found 0 quests in 20ms
AddCampType: Found 0 quests in 27ms
AddCampPower: Found 0 quests in 16ms
AddCampShower: Found 0 quests in 37ms
AddToiletsFee: Found 0 quests in 31ms
AddBabyChangingTable: Found 1 quests in 34ms
AddSmoking: Found 7 quests in 234ms
AddWheelchairAccessToiletsPart: Found 0 quests in 17ms
AddWheelchairAccessToilets: Found 0 quests in 18ms
AddBikeRepairAvailability: Found 1 quests in 20ms
AddSecondHandBicycleAvailability: Found 1 quests in 18ms
AddForestLeafType: Found 0 quests in 353ms
AddToiletAvailability: Found 0 quests in 105ms
AddKosher: Found 10 quests in 89ms
AddHalal: Found 10 quests in 101ms
AddWheelchairAccessBusiness: Found 28 quests in 88ms
AddVegetarian: Found 7 quests in 129ms
AddAcceptsCards: Found 17 quests in 27ms
AddInternetAccess: Found 3 quests in 110ms
AddVegan: Found 1 quests in 169ms
AddTracktype: Found 0 quests in 82ms
AddLanes: Found 0 quests in 286ms
AddSidewalk: Found 11 quests in 350ms
AddRoadSmoothness: Found 178 quests in 266ms
AddRoadWidth: Found 63 quests in 422ms
AddFootwayPartSurface: Found 0 quests in 124ms
AddPathSmoothness: Found 298 quests in 313ms
CheckShopExistence: Found 3 quests in 693ms
AddSidewalkSurface: Found 14 quests in 33ms
AddCyclewayPartSurface: Found 0 quests in 93ms
AddMaxSpeed: Found 3 quests in 22ms
AddCyclewayWidth: Found 8 quests in 32ms
AddCyclewaySegregation: Found 0 quests in 385ms
AddStepCount: Found 0 quests in 16ms
AddBuildingType: Found 52 quests in 54ms
AddBuildingLevels: Found 738 quests in 62ms
**AddIsAmenityIndoor: Found 4 quests in 1156ms**
AddRoofShape: Found 628 quests in 353ms
AddMaxHeight: Found 0 quests in 1818ms
AddRoadSurface: Found 0 quests in 1140ms
AddPathSurface: Found 13 quests in 1200ms
AddWayLit: Found 128 quests in 1128ms
</details>



**Around Pedestrian Area / Railway Station**
AddIsAmenityIndoor: Found 45 quests in 4406ms
For comparison: average for all quests was 530ms, only CheckExistence and AddOpeningHours were slower

<details>
<summary>Log  - 41239 nodes, 8433 ways</summary>

AddBusStopShelter: Found 8 quests in 356ms
AddSuspectedOneway: Found 0 quests in 478ms
AddPlaceName: Found 4 quests in 782ms
AddBenchStatusOnBusStop: Found 10 quests in 435ms
AddBusStopName: Found 17 quests in 106ms
AddBinStatusOnBusStop: Found 13 quests in 488ms
AddTactilePavingBusStop: Found 4 quests in 494ms
AddCarWashType: Found 0 quests in 72ms
AddBenchBackrest: Found 5 quests in 127ms
AddAmenityCover: Found 1 quests in 67ms
AddBridgeStructure: Found 0 quests in 63ms
MarkCompletedBuildingConstruction: Found 2 quests in 71ms
AddSport: Found 0 quests in 39ms
AddHousenumber: Found 19 quests in 1215ms
AddPitchSurface: Found 1 quests in 60ms
AddRailwayCrossingBarrier: Found 0 quests in 394ms
AddParkingType: Found 0 quests in 103ms
AddPitchLit: Found 1 quests in 108ms
AddParkingAccess: Found 0 quests in 80ms
AddBusStopLit: Found 9 quests in 511ms
AddTrafficCalmingType: Found 0 quests in 33ms
AddStepsIncline: Found 10 quests in 43ms
AddParkingFee: Found 0 quests in 248ms
AddBicycleIncline: Found 0 quests in 198ms
AddMemorialType: Found 1 quests in 70ms
AddReligionToPlaceOfWorship: Found 0 quests in 78ms
AddReligionToWaysideShrine: Found 0 quests in 51ms
AddPowerPolesMaterial: Found 0 quests in 45ms
AddIsBuildingUnderground: Found 0 quests in 50ms
AddMotorcycleParkingCover: Found 0 quests in 56ms
AddAirCompressor: Found 0 quests in 52ms
AddMotorcycleParkingCapacity: Found 1 quests in 107ms
AddRecyclingType: Found 0 quests in 50ms
DetermineRecyclingGlass: Found 0 quests in 50ms
AddTactilePavingSteps: Found 107 quests in 585ms
AddRecyclingContainerMaterials: Found 4 quests in 343ms
AddCycleway: Found 117 quests in 2328ms
AddCrossing: Found 1 quests in 414ms
AddCrossingIsland: Found 0 quests in 406ms
AddHandrail: Found 7 quests in 1573ms
AddStepsRamp: Found 15 quests in 1606ms
AddCrossingType: Found 1 quests in 726ms
AddTrafficSignalsButton: Found 3 quests in 45ms
AddCrossingKerbHeight: Found 11 quests in 516ms
AddInformationToTourism: Found 0 quests in 102ms
AddPlaygroundAccess: Found 0 quests in 69ms
AddTactilePavingCrosswalk: Found 1 quests in 1094ms
AddGritBinSeasonal: Found 27 quests in 44ms
AddBoardType: Found 0 quests in 42ms
AddPostboxCollectionTimes: Found 6 quests in 347ms
AddBarrierType: Found 0 quests in 26ms
AddBarrierOnRoad: Found 0 quests in 174ms
AddBicycleBarrierType: Found 0 quests in 23ms
AddTactilePavingKerb: Found 4 quests in 2241ms
AddTrafficSignalsSound: Found 3 quests in 1256ms
AddKerbHeight: Found 1 quests in 2327ms
AddBarrierOnPath: Found 1 quests in 253ms
AddBicycleBarrierInstallation: Found 1 quests in 59ms
AddSelfServiceLaundry: Found 0 quests in 55ms
AddGeneralFee: Found 7 quests in 66ms
AddDrinkingWater: Found 1 quests in 68ms
AddDrinkingWaterType: Found 0 quests in 38ms
AddTrafficSignalsVibration: Found 3 quests in 1069ms
AddCameraType: Found 0 quests in 23ms
AddSeating: Found 353 quests in 319ms
AddStepCountStile: Found 0 quests in 340ms
AddFireHydrantPosition: Found 7 quests in 86ms
AddFireHydrantDiameter: Found 15 quests in 46ms
AddFireHydrantType: Found 0 quests in 126ms
AddBikeParkingCover: Found 0 quests in 89ms
AddBikeParkingType: Found 2 quests in 62ms
AddBikeRentalType: Found 2 quests in 75ms
AddStileType: Found 0 quests in 535ms
AddBikeParkingAccess: Found 0 quests in 124ms
AddBikeParkingFee: Found 0 quests in 102ms
AddBollardType: Found 13 quests in 539ms
AddBikeRentalCapacity: Found 0 quests in 109ms
SpecifyShopType: Found 2 quests in 73ms
AddHairdresserCustomers: Found 41 quests in 88ms
AddAtmOperator: Found 1 quests in 29ms
AddAtmCashIn: Found 3 quests in 29ms
AddBicyclePump: Found 1 quests in 90ms
AddChargingStationCapacity: Found 0 quests in 74ms
AddChargingStationOperator: Found 0 quests in 54ms
AddAccessPointRef: Found 0 quests in 31ms
AddWheelchairAccessOutside: Found 0 quests in 51ms
AddClothingBinOperator: Found 0 quests in 243ms
AddMaxWeight: Found 0 quests in 33ms
AddBikeParkingCapacity: Found 8 quests in 606ms
AddRoadName: Found 0 quests in 44ms
AddAddressStreet: Found 1 quests in 583ms
AddOneway: Found 0 quests in 194ms
AddMaxPhysicalHeight: Found 3 quests in 338ms
AddFerryAccessMotorVehicle: Found 0 quests in 33ms
AddFerryAccessPedestrian: Found 0 quests in 48ms
CheckOpeningHoursSigned: Found 27 quests in 816ms
AddProhibitedForPedestrians: Found 0 quests in 51ms
MarkCompletedHighwayConstruction: Found 0 quests in 47ms
CheckExistence: Found 28 quests in 5702ms
AddOrchardProduce: Found 0 quests in 35ms
AddEntrance: Found 4 quests in 540ms
AddSummitCross: Found 0 quests in 453ms
AddSummitRegister: Found 0 quests in 497ms
CheckShopType: Found 32 quests in 1382ms
AddLevel: Found 0 quests in 406ms
AddAirConditioning: Found 437 quests in 298ms
AddCampDrinkingWater: Found 0 quests in 47ms
AddForestLeafType: Found 0 quests in 638ms
AddCampType: Found 0 quests in 92ms
AddCampShower: Found 0 quests in 66ms
AddCampPower: Found 0 quests in 45ms
AddToiletsFee: Found 1 quests in 56ms
AddBabyChangingTable: Found 30 quests in 101ms
AddWheelchairAccessToiletsPart: Found 44 quests in 112ms
AddBikeRepairAvailability: Found 2 quests in 52ms
AddToiletAvailability: Found 1 quests in 266ms
AddSecondHandBicycleAvailability: Found 3 quests in 46ms
AddWheelchairAccessToilets: Found 3 quests in 167ms
AddWheelchairAccessPublicTransport: Found 2 quests in 782ms
AddHalal: Found 338 quests in 524ms
AddMaxHeight: Found 16 quests in 1946ms
AddSmoking: Found 245 quests in 1070ms
AddVegan: Found 60 quests in 577ms
AddVegetarian: Found 213 quests in 697ms
AddAcceptsCards: Found 721 quests in 160ms
AddWheelchairAccessBusiness: Found 609 quests in 203ms
AddKosher: Found 343 quests in 418ms
AddTracktype: Found 0 quests in 58ms
AddInternetAccess: Found 39 quests in 600ms
AddLanes: Found 10 quests in 497ms
AddRoadWidth: Found 300 quests in 598ms
AddSidewalk: Found 126 quests in 872ms
AddRoadSmoothness: Found 656 quests in 693ms
AddFootwayPartSurface: Found 4 quests in 268ms
AddCyclewaySegregation: Found 2 quests in 593ms
AddSidewalkSurface: Found 195 quests in 45ms
AddCyclewayWidth: Found 72 quests in 71ms
AddPathSmoothness: Found 775 quests in 1045ms
AddMaxSpeed: Found 4 quests in 36ms
AddBuildingLevels: Found 760 quests in 52ms
AddBuildingType: Found 475 quests in 67ms
AddStepCount: Found 11 quests in 48ms
AddCyclewayPartSurface: Found 4 quests in 343ms
AddOpeningHours: Found 375 quests in 9087ms
AddRoofShape: Found 798 quests in 446ms
AddRoadSurface: Found 18 quests in 2330ms
CheckShopExistence: Found 170 quests in 3160ms
AddPathSurface: Found 24 quests in 2468ms
**AddIsAmenityIndoor: Found 45 quests in 4406ms**
AddWayLit: Found 146 quests in 2815ms
Created 9047 quests for bbox in 12.0s
</details>

Reply:
This looks acceptable to me, but at the same time it slow enough that it's worth trying some performance improvements.

Two things that may help:
* Do you really need the list of buildings for anything other than getting geometries? If not, you could just iterate over the geometries (without caring about the associated `id`, so also no map necessary).
* Before checking `isInMultipolygon`, a `node.position in geometry.getBounds()` check could be done. This should be considerably faster as it will reduce number of calls to the slower `isInMultipolygon`.

Reply:
For performance improvement, have a look at what is being done in `AddHousenumber`. See the `LatLonRaster`. The same can be done here:
1. sort the positions of the amenity-nodes into the raster
2. for each building, only make point in polygon checks for the nearby points

Reply:
>     * Before checking `isInMultipolygon`, a `node.position in geometry.getBounds()` check could be done. This should be considerably faster as it will reduce number of calls to the slower `isInMultipolygon`.
Yes, this makes sense to me. Checking if LatLon is in Bounds is way faster than point-in-polygon algorythtm. I added this in my latest commit.



> For performance improvement, have a look at what is being done in `AddHousenumber`. See the `LatLonRaster`. The same can be done here:
> 
>     1. sort the positions of the amenity-nodes into the raster
> 
>     2. for each building, only make point in polygon checks for the nearby points
I'm not sure, if i understood this right, but i added a LatLonRaster and used this to remove all buildings with no nodes in their bounds. So we have to iterate over less buildings later.

After these [two changes](https://github.com/qugebert/StreetComplete/commit/3a5932f996a89bb9b895370c1fb3c58c13cea6cf) it seems faster
AddIsAmenityIndoor: Found 40 quests in 591ms

<details>
<summary>Example Log around Pedestrian Area / Railway Station</summary>

AddBenchStatusOnBusStop: Found 10 quests in 481ms
AddBusStopShelter: Found 8 quests in 509ms
AddSuspectedOneway: Found 0 quests in 516ms
AddBusStopName: Found 17 quests in 138ms
AddBinStatusOnBusStop: Found 13 quests in 399ms
AddTactilePavingBusStop: Found 4 quests in 422ms
AddPlaceName: Found 4 quests in 955ms
AddCarWashType: Found 0 quests in 53ms
AddBenchBackrest: Found 5 quests in 79ms
AddAmenityCover: Found 1 quests in 40ms
AddBusStopLit: Found 9 quests in 385ms
AddBridgeStructure: Found 0 quests in 39ms
MarkCompletedBuildingConstruction: Found 2 quests in 61ms
AddSport: Found 0 quests in 65ms
AddPitchSurface: Found 1 quests in 71ms
AddRailwayCrossingBarrier: Found 0 quests in 276ms
AddPitchLit: Found 1 quests in 78ms
AddParkingType: Found 0 quests in 75ms
AddTrafficCalmingType: Found 0 quests in 20ms
AddParkingAccess: Found 0 quests in 76ms
AddStepsIncline: Found 10 quests in 34ms
AddParkingFee: Found 0 quests in 197ms
AddHousenumber: Found 19 quests in 1441ms
AddBicycleIncline: Found 0 quests in 105ms
AddMemorialType: Found 1 quests in 48ms
AddReligionToWaysideShrine: Found 0 quests in 48ms
AddReligionToPlaceOfWorship: Found 0 quests in 69ms
AddPowerPolesMaterial: Found 0 quests in 57ms
AddIsBuildingUnderground: Found 0 quests in 54ms
AddMotorcycleParkingCover: Found 0 quests in 73ms
AddMotorcycleParkingCapacity: Found 1 quests in 110ms
AddAirCompressor: Found 0 quests in 54ms
AddTactilePavingSteps: Found 107 quests in 505ms
AddRecyclingType: Found 0 quests in 52ms
DetermineRecyclingGlass: Found 0 quests in 57ms
AddRecyclingContainerMaterials: Found 4 quests in 281ms
AddCycleway: Found 117 quests in 2237ms
AddCrossing: Found 1 quests in 348ms
AddCrossingIsland: Found 0 quests in 229ms
AddStepsRamp: Found 15 quests in 1490ms
AddHandrail: Found 7 quests in 1523ms
AddCrossingKerbHeight: Found 10 quests in 265ms
AddTrafficSignalsButton: Found 3 quests in 31ms
AddCrossingType: Found 1 quests in 618ms
AddInformationToTourism: Found 0 quests in 85ms
AddPlaygroundAccess: Found 0 quests in 57ms
AddTactilePavingCrosswalk: Found 1 quests in 785ms
AddGritBinSeasonal: Found 27 quests in 37ms
AddBoardType: Found 0 quests in 27ms
AddBarrierType: Found 0 quests in 28ms
AddPostboxCollectionTimes: Found 6 quests in 270ms
AddBarrierOnPath: Found 1 quests in 191ms
AddKerbHeight: Found 1 quests in 1808ms
AddBicycleBarrierType: Found 0 quests in 39ms
AddTactilePavingKerb: Found 4 quests in 1816ms
AddBarrierOnRoad: Found 0 quests in 181ms
AddBicycleBarrierInstallation: Found 1 quests in 148ms
AddStileType: Found 0 quests in 147ms
AddSelfServiceLaundry: Found 0 quests in 61ms
AddStepCountStile: Found 0 quests in 220ms
AddTrafficSignalsSound: Found 3 quests in 1108ms
AddBollardType: Found 13 quests in 228ms
AddCameraType: Found 0 quests in 32ms
AddFireHydrantType: Found 0 quests in 26ms
AddDrinkingWater: Found 1 quests in 86ms
AddGeneralFee: Found 7 quests in 124ms
AddFireHydrantPosition: Found 7 quests in 49ms
AddFireHydrantDiameter: Found 15 quests in 47ms
AddDrinkingWaterType: Found 0 quests in 129ms
AddTrafficSignalsVibration: Found 3 quests in 997ms
AddBikeParkingCover: Found 0 quests in 82ms
AddSeating: Found 353 quests in 281ms
AddBikeRentalType: Found 2 quests in 52ms
AddBikeParkingType: Found 2 quests in 56ms
AddHairdresserCustomers: Found 40 quests in 65ms
AddBikeRentalCapacity: Found 0 quests in 79ms
AddBikeParkingAccess: Found 0 quests in 119ms
AddBikeParkingFee: Found 0 quests in 106ms
SpecifyShopType: Found 2 quests in 67ms
AddAtmOperator: Found 1 quests in 42ms
AddAddressStreet: Found 1 quests in 200ms
AddAtmCashIn: Found 3 quests in 29ms
AddBicyclePump: Found 1 quests in 125ms
AddClothingBinOperator: Found 0 quests in 70ms
AddChargingStationCapacity: Found 0 quests in 59ms
AddChargingStationOperator: Found 0 quests in 78ms
AddAccessPointRef: Found 0 quests in 29ms
AddMaxWeight: Found 0 quests in 52ms
AddWheelchairAccessOutside: Found 0 quests in 80ms
AddRoadName: Found 0 quests in 46ms
AddMaxPhysicalHeight: Found 3 quests in 259ms
AddOneway: Found 0 quests in 176ms
AddFerryAccessPedestrian: Found 0 quests in 58ms
AddBikeParkingCapacity: Found 10 quests in 700ms
AddFerryAccessMotorVehicle: Found 0 quests in 34ms
MarkCompletedHighwayConstruction: Found 0 quests in 47ms
AddProhibitedForPedestrians: Found 0 quests in 91ms
CheckOpeningHoursSigned: Found 27 quests in 880ms
AddEntrance: Found 4 quests in 358ms
AddOrchardProduce: Found 0 quests in 36ms
AddSummitCross: Found 0 quests in 237ms
AddSummitRegister: Found 0 quests in 243ms
AddLevel: Found 0 quests in 167ms
AddForestLeafType: Found 0 quests in 253ms
AddAirConditioning: Found 437 quests in 264ms
CheckShopType: Found 31 quests in 1232ms
AddCampType: Found 0 quests in 55ms
AddCampDrinkingWater: Found 0 quests in 58ms
AddCampShower: Found 0 quests in 56ms
AddCampPower: Found 0 quests in 55ms
AddToiletsFee: Found 1 quests in 75ms
AddBabyChangingTable: Found 30 quests in 139ms
AddToiletAvailability: Found 1 quests in 283ms
AddWheelchairAccessToilets: Found 3 quests in 116ms
**AddIsAmenityIndoor: Found 40 quests in 591ms**
AddMaxHeight: Found 16 quests in 1511ms
CheckExistence: Found 28 quests in 5845ms
AddWheelchairAccessToiletsPart: Found 44 quests in 180ms
AddSecondHandBicycleAvailability: Found 3 quests in 52ms
AddBikeRepairAvailability: Found 2 quests in 87ms
AddWheelchairAccessPublicTransport: Found 2 quests in 876ms
AddWheelchairAccessBusiness: Found 609 quests in 168ms
AddHalal: Found 338 quests in 271ms
AddKosher: Found 343 quests in 294ms
AddSmoking: Found 245 quests in 1163ms
AddAcceptsCards: Found 721 quests in 144ms
AddTracktype: Found 0 quests in 38ms
AddVegan: Found 60 quests in 437ms
AddInternetAccess: Found 39 quests in 480ms
AddVegetarian: Found 213 quests in 748ms
AddSidewalk: Found 124 quests in 538ms
AddRoadWidth: Found 300 quests in 491ms
AddLanes: Found 10 quests in 590ms
AddFootwayPartSurface: Found 4 quests in 380ms
AddRoadSmoothness: Found 656 quests in 833ms
AddSidewalkSurface: Found 196 quests in 52ms
AddCyclewaySegregation: Found 2 quests in 749ms
AddCyclewayWidth: Found 72 quests in 71ms
AddCyclewayPartSurface: Found 4 quests in 332ms
AddMaxSpeed: Found 4 quests in 63ms
AddBuildingType: Found 474 quests in 90ms
AddPathSmoothness: Found 776 quests in 1060ms
AddBuildingLevels: Found 761 quests in 76ms
AddStepCount: Found 11 quests in 56ms
AddRoofShape: Found 799 quests in 285ms
AddRoadSurface: Found 18 quests in 2356ms
AddOpeningHours: Found 375 quests in 8784ms
CheckShopExistence: Found 170 quests in 3074ms
AddPathSurface: Found 23 quests in 2560ms
AddWayLit: Found 146 quests in 2978ms

Average: 461.88 ms

</details>

Reply:
Wow cool, so 8 times faster or so :-)